src/deluge/gui/menu_item/audio_compressor/compressor_params.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2014-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include "definitions_cxx.hpp"
+#include "gui/menu_item/unpatched_param.h"
+#include "gui/ui/sound_editor.h"
+#include "modulation/params/param_set.h"
+
+namespace deluge::gui::menu_item::audio_compressor {
+
+class CompParam final : public UnpatchedParam {
+public:
+	using UnpatchedParam::UnpatchedParam;
+	void readCurrentValue() override {
+		auto value = (int64_t)soundEditor.currentParamManager->getUnpatchedParamSet()->getValue(getP());
+		this->setValue((value * (kMaxMenuValue * 2) + 2147483648) >> 32);
+	}
+	// comp params aren't set up for negative inputs - this is the same as osc pulse width
```

Comment:
Sadly threshold is unpatched and oscPulseWidth is patched, I couldn't think of a good way to do it without making more seperate classes (e.g. they'dbe IntegerPatchedNonFMPositive and IntegerUnpatchedPositive) since the patched and unpatched versions need to call different functions to get their param collection

Happy to do it if I've missed something